packages/stitch/tests/stitchSchemasPathBug.test.ts

```diff
@@ -0,0 +1,96 @@
+import { describe, expect, test } from 'vitest';
+import { makeExecutableSchema } from '@graphql-tools/schema';
+import { stitchSchemas } from '@graphql-tools/stitch';
+import { graphql, OperationTypeNode } from 'graphql';
+import { delegateToSchema } from '@graphql-tools/delegate';
+
+describe('Supergraph handles stitched field with different name correctly', () => {
+    test('ensures correct error path in response', async () => {
+        const subschema = makeExecutableSchema({
+            typeDefs: /* GraphQL */ `
+            type Query {
+                books(ids: [ID!]!): [Book!]!
+            }
+
+            type Book {
+                id: ID!
+                title: String
+                summary: String
+            }
+            `,
+            resolvers: {
+                Query: {
+                    books: (_parent, args) => ([{
+                        id: args.id,
+                        title: () => {
+                            throw new Error('Title - Forced error for testing purposes');
+                        },
+                        summary: () => {
+                            throw new Error('Summary - Forced error for testing purposes');
+                        },
+                    }]),
+                },
+            },
+        });
+
+        // Stitch the subschema into the supergraph
+        const supergraphSchema = stitchSchemas({
+            subschemas: [subschema],
+            typeDefs: /* GraphQL */ `
+                type Query {
+                    book(id: ID!): Book
+                }
+
+                type Book {
+                    id: ID!
+                    title: String
+                    summary: String
+                }
+            `,
+            resolvers: {
+                Query: {
+                    book: {
+                        selectionSet: '{ id }',
+                        resolve: (_parent, args, context, info) => {
+                            const test =  delegateToSchema({
+                                schema: subschema,
+                                operation: 'query' as OperationTypeNode,
+                                fieldName: 'books',
+                                args: {
+                                    ids: [args.id],
+                                },
+                                context,
+                                info,
+                            });
+                            return test[0];
```

Comment:
I think this is why you get the wrong path. Because you transform the data on your own here.
Maybe you should use batch delegation with `batchDelegateToSchema` from `@graphql-tools/batch-delegate` or 
use transforms to handle data transformation;
https://the-guild.dev/graphql/stitching/docs/transforms/operational

Because delegateToSchema expects you have the same return type.

Reply:
Thanks @ardatan for your quick reply! 

Our actual setup is a little different from what I had in the tests originally. We are stitching schemas and using `merge` to resolve the fields. I've pushed an update to this test: [`dcd6b42` (#742)](https://github.com/graphql-hive/gateway/pull/742/commits/dcd6b4286e0d7f0d1b51548388826cc0478dab9f). This is truer to what we're seeing in our setup.